numpy/linalg/linalg.py

```diff
@@ -2544,30 +2776,14 @@ def norm(x, ord=None, axis=None, keepdims=False):
         axis = (axis,)
 
     if len(axis) == 1:
-        if ord == Inf:
-            return abs(x).max(axis=axis, keepdims=keepdims)
-        elif ord == -Inf:
-            return abs(x).min(axis=axis, keepdims=keepdims)
-        elif ord == 0:
-            # Zero norm
-            return (x != 0).astype(x.real.dtype).sum(axis=axis, keepdims=keepdims)
-        elif ord == 1:
-            # special case for speedup
-            return add.reduce(abs(x), axis=axis, keepdims=keepdims)
+        ret = sqnorm(x, ord=ord, axis=axis, keepdims=keepdims)
+        if ord in (Inf, -Inf, 0, 1):
+            return ret
         elif ord is None or ord == 2:
             # special case for speedup
-            s = (x.conj() * x).real
-            return sqrt(add.reduce(s, axis=axis, keepdims=keepdims))
-        # None of the str-type keywords for ord ('fro', 'nuc')
-        # are valid for vectors
-        elif isinstance(ord, str):
-            raise ValueError(f"Invalid norm order '{ord}' for vectors")
+            return sqrt(ret)
         else:
-            absx = abs(x)
-            absx **= ord
-            ret = add.reduce(absx, axis=axis, keepdims=keepdims)
-            ret **= (1 / ord)
-            return ret
+            return ret ** (1 / ord)
```

Comment:
`norm_sq` feels like a weird name when `ord != 2` for this case, although I also can't really think of a better name.